lab2/kursinis.tex

```diff
@@ -47,10 +47,29 @@ \section{Context}
 the educational experience for both students and teachers. With a user-friendly interface and a comprehensive set of features, EduPal aims to streamline the learning process, fostering a more efficient and engaging educational environment.
 
 \subsection{Functionality overview}
-List features here
 
-\subsection{Tech stack}
-List tech stack here
+At present, EduPal has these functionalities:
+
+\begin{itemize}
+    \item \textbf{Create and Manage Subjects}: Users have the ability to create new subjects and manage existing ones within the system. This includes creating topics under each subject and uploading notes in PDF format related to those topics. Additionally, users can conveniently access all their own notes as well as notes shared by peers within the system.
+    \item \textbf{Note Sharing and Interaction}: Users can interact with notes shared by others by viewing, ranking, and engaging in discussions within the topic's section. The system allows users to rank notes based on usefulness to promote the most valuable ones. Users also have options to export notes to a PDF section, download them to their device, or save them as drafts for further editing. 
+    \item \textbf{Pomodoro Timer System}: The system features a Pomodoro timer system designed to enhance productivity. Users can customize their study and break times according to their preferences. The timer notifies users when it's time to take a break, helping them maintain focus and manage their study sessions effectively.
+    \item \textbf{Goal Setting}: Users can set study goals within the system, specifying desired subjects and study times. This feature enables users to plan and organize their study sessions more efficiently, helping them stay on track with their academic or learning objectives.
+    \item \textbf{Integration with GPT API}: The project integrates with Chat GPT, providing users with additional functionalities. This integration allows the system to store user chat history and respond to user inquiries using the capabilities of Chat GPT. Users can leverage this feature to seek assistance, clarify doubts, or engage in interactive conversations within the system.
+    \end{itemize}
+
+\subsection{Tech stack and services}
+
+The EduPal system was created using a modern technology stack to support its functionalities. The basic structure of the system:
+
+\begin{itemize}
+    \item \textbf{Frontend}: The frontend of the system is developed using the React framework and JavaScript. Main component libraries were ej2-react-dropdowns, which provides UI components and specifically refers to dropdown components in React, and react-pdf, which allows the display of PDF documents. Frontend allows communication with server via fetch function which sends the HTTP request to the particular API endpoint in the backend. After the backend sends HTTP response to the frontend, React components are updated accordingly, mainly thorugh setting component state or triggering UI updates.
```

Comment:
No need to provide any libraries. No UI library is used, instead projects uses custom CSS. "After the backend sends HTTP response to the frontend, React components are updated accordingly, mainly thorugh setting component state or triggering UI updates." - useless sentence, you had to specify how requests are made (fetch api, axios)